docs/data/tree-view/rich-tree-view/selection/selection.md

```diff
@@ -75,25 +75,36 @@ Use the `onItemSelectionToggle` prop if you want to react to an item selection c
 
 {{"demo": "TrackItemSelectionToggle.js"}}
 
-## Parent / children selection relationship
+## Automatic parents and children selection
 
-Automatically select an item when all of its children are selected and automatically select all children when the parent is selected.
+By default, selecting a parent item does not select its children. You can override this behavior using the `selectionPropagation` prop.
 
-:::warning
-This feature isn't implemented yet. It's coming.
+Here's how it's structured:
 
-👍 Upvote [issue #4821](https://github.com/mui/mui-x/issues/4821) if you want to see it land faster.
+```ts
+type TreeViewSelectionPropagation = {
+  descendants?: boolean; // default: false
+  parents?: boolean; // default: false
+};
+```
 
-Don't hesitate to leave a comment on the same issue to influence what gets built.
-Especially if you already have a use case for this component,
-or if you are facing a pain point with your current solution.
-:::
+When `rowSelectionPropagation.descendants` is set to `true`.
+
+- Selecting a parent selects all its filtered descendants automatically.
+- Deselecting a parent row deselects all its filtered descendants automatically.
 
-If you cannot wait for the official implementation,
-you can create your own custom solution using the `selectedItems`,
-`onSelectedItemsChange` and `onItemSelectionToggle` props:
+When `rowSelectionPropagation.parents` is set to `true`.
 
-{{"demo": "ParentChildrenSelectionRelationship.js"}}
+- Selecting all the filtered descendants of a parent selects the parent automatically.
+- Deselecting a descendant of a selected parent deselects the parent automatically.
+
+The example below demonstrates the usage of the `selectionPropagation` prop.
+
+{{"demo": "SelectionPropagation.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
```suggestion
{{"demo": "SelectionPropagation.js", "defaultCodeOpen": false}}
```
I think the demo looks a bit weird without the border because of the inconsistent borderRadius of the toolbar here 🙈 
![image](https://github.com/user-attachments/assets/fc2ddc20-f75c-45cc-971a-137a70c3cca8)